src/ompl/geometric/planners/rrt/src/RRTstar.cpp

```diff
@@ -581,7 +603,8 @@ ompl::base::PlannerStatus ompl::geometric::RRTstar::solve(const base::PlannerTer
                 "%.3f",
                 getName().c_str(), statesGenerated, rewireTest, goalMotions_.size(), bestCost_.value());
 
-    return base::PlannerStatus(addedSolution, approximate);
+    // We've added a solution if newSolution == true, and it is an approximate solution if bestGoalMotion_ == false
+    return base::PlannerStatus(static_cast<bool>(newSolution), !static_cast<bool>(bestGoalMotion_));
```

Comment:
This is more of a style comment. I think this would be better:
```
return base::PlannerStatus(newSolution != nullptr, bestGoalMotion_ == nullptr);
```